src/core/highlight.js

```diff
@@ -45,32 +45,44 @@ export async function run(conf) {
         console.error("Timed-out waiting for highlight:", element);
         done();
       }, 4000);
-      const msg = {
-        action: "highlight",
-        code: element.textContent,
-        id: `highlight:${i}`,
-        languages: getLanguageHint(element.classList),
-      };
-      worker.addEventListener("message", function listener(ev) {
-        const {
-          data: { id, language, value },
-        } = ev;
-        if (id !== msg.id) {
-          return; // not for us!
-        }
-        element.innerHTML = value;
-        if (element.localName === "pre") {
-          element.classList.add("hljs");
-        }
-        if (language) {
-          element.classList.add(language);
-        }
-        clearTimeout(timeoutId);
-        worker.removeEventListener("message", listener);
-        done();
+      const elementsToHighlight =
+        element.querySelectorAll("code").length > 0
+          ? Array.from(element.querySelectorAll("code"))
+          : [element];
+      elementsToHighlight.map((each, id) => {
```

Comment:
As convention, when we work with elements, we call them "elem" so we know what we are working with.  

```suggestion
      elementsToHighlight.map((elem, id) => {
```

Please change this bellow too.